quickstarts/microsoft.compute/1-vm-loadbalancer-2-nics/main.bicep

```diff
@@ -0,0 +1,293 @@
+  uefiSettings: {
+    secureBootEnabled: true
+    vTpmEnabled: true
+  }
+  securityType: securityType
+}
+var addressPrefix = '10.0.0.0/16'
+var imageOffer = 'WindowsServer'
+var imagePublisher = 'MicrosoftWindowsServer'
+var imageSKU = OSVersion
+var lbName = 'myLB'
+var networkSecurityGroupName = '${subnetName}-nsg'
+var nic1Name = 'nic1'
+var nic2Name = 'nic2'
+var publicIPAddressName = 'myPublicIP'
+var publicIPAddressType = 'Dynamic'
+var storageAccountType = 'Standard_LRS'
+var subnetName = 'Subnet-1'
+var subnetPrefix = '10.0.0.0/24'
+var vmName = 'myVM'
+var vnetName = 'myVNET'
+
+resource storageAccount 'Microsoft.Storage/storageAccounts@2023-01-01' = {
+  name: storageAccountName
+  location: location
+  sku: {
+    name: storageAccountType
+  }
+  kind: 'StorageV2'
+}
+
+resource lb 'Microsoft.Network/loadBalancers@2023-05-01' = {
+  name: lbName
+  location: location
+  properties: {
+    frontendIPConfigurations: [
+      {
+        name: 'LoadBalancerFrontEnd'
+        properties: {
+          publicIPAddress: {
+            id: publicIPAddress.id
+          }
+        }
+      }
+    ]
+    backendAddressPools: [
+      {
+        name: 'BackendPool1'
+      }
+    ]
+    inboundNatRules: [
+      {
+        name: 'RDP-VM0'
+        properties: {
+          frontendIPConfiguration: {
+            id: resourceId('Microsoft.Network/loadBalancers/frontendIPConfigurations', lbName, 'LoadBalancerFrontEnd')
+          }
+          protocol: 'Tcp'
+          frontendPort: 50001
+          backendPort: 3389
+          enableFloatingIP: false
+        }
+      }
+    ]
+  }
+}
+
+resource networkSecurityGroup 'Microsoft.Network/networkSecurityGroups@2023-05-01' = {
+  name: networkSecurityGroupName
+  location: location
+  properties: {}
+}
+
+resource nic1 'Microsoft.Network/networkInterfaces@2023-05-01' = {
+  name: nic1Name
+  location: location
+  properties: {
+    ipConfigurations: [
+      {
+        name: 'ipconfig1'
+        properties: {
+          privateIPAllocationMethod: 'Dynamic'
+          subnet: {
+            id: resourceId('Microsoft.Network/virtualNetworks/subnets', vnetName, subnetName)
+          }
+          loadBalancerBackendAddressPools: [
+            {
+              id: resourceId('Microsoft.Network/loadBalancers/backendAddressPools', lbName, 'BackendPool1')
+            }
+          ]
+          loadBalancerInboundNatRules: [
+            {
+              id: resourceId('Microsoft.Network/loadBalancers/inboundNatRules', lbName, 'RDP-VM0')
+            }
+          ]
+        }
+      }
+    ]
+  }
+  dependsOn: [
+    vnet
+    lb
+  ]
+}
+
+resource nic2 'Microsoft.Network/networkInterfaces@2023-05-01' = {
+  name: nic2Name
+  location: location
+  properties: {
+    ipConfigurations: [
+      {
+        name: 'ipconfig1'
+        properties: {
+          privateIPAllocationMethod: 'Dynamic'
+          subnet: {
+            id: resourceId('Microsoft.Network/virtualNetworks/subnets', vnetName, subnetName)
```

Comment:
Rather than using `resourceId()` + `dependsOn`, I would recommend instead doing:

```bicep
resource subnet 'Microsoft.Network/virtualNetworks/subnets@2023-05-01' existing = {
  parent: vnet
  name: subnetName
}
```

And then referencing this resource with:
```bicep
subnet: {
  id: subnet.id
}
```